### ISSUE TITLE
使用自定义数据集问题

### ISSUE THREAD

**Issue by caomao1111** [PaddlePaddle/PaddleSeg#2125]
使用BiseNetV2训练我自己的裂缝数据集（二分类的分割任务）训练，结果mIoU永远为0.5是为何，训练loss一直在降低，但mIoU完全没变，数据集是灰度图

**Comment by shiyutang** [comment]
Can you provide the information about the loss change?

**Comment by caomao1111** [comment]
2022-05-19 10:36:53 [INFO]      [TRAIN] epoch: 2, iter: 110/1000, loss: 0.0006, lr: 0.045067, batch_cost: 0.1060, reader_cost: 0.00008, ips: 37.7453 samples/sec | ETA 00:01:34
2022-05-19 10:36:54 [INFO]      [TRAIN] epoch: 2, iter: 120/1000, loss: 0.0007, lr: 0.044612, batch_cost: 0.1000, reader_cost: 0.00006, ips: 40.0111 samples/sec | ETA 00:01:27
2022-05-19 10:36:55 [INFO]      [TRAIN] epoch: 2, iter: 130/1000, loss: 0.0006, lr: 0.044156, batch_cost: 0.1001, reader_cost: 0.00006, ips: 39.9770 samples/sec | ETA 00:01:27
2022-05-19 10:36:56 [INFO]      [TRAIN] epoch: 2, iter: 140/1000, loss: 0.0006, lr: 0.043699, batch_cost: 0.1009, reader_cost: 0.00006, ips: 39.6319 samples/sec | ETA 00:01:26
2022-05-19 10:36:57 [INFO]      [TRAIN] epoch: 2, iter: 150/1000, loss: 0.0005, lr: 0.043242, batch_cost: 0.0974, reader_cost: 0.00006, ips: 41.0819 samples/sec | ETA 00:01:22
2022-05-19 10:36:58 [INFO]      [TRAIN] epoch: 3, iter: 160/1000, loss: 0.0006, lr: 0.042784, batch_cost: 0.1154, reader_cost: 0.00751, ips: 34.6560 samples/sec | ETA 00:01:36
2022-05-19 10:36:59 [INFO]      [TRAIN] epoch: 3, iter: 170/1000, loss: 0.0006, lr: 0.042326, batch_cost: 0.1035, reader_cost: 0.00006, ips: 38.6371 samples/sec | ETA 00:01:25
2022-05-19 10:37:00 [INFO]      [TRAIN] epoch: 3, iter: 180/1000, loss: 0.0006, lr: 0.041868, batch_cost: 0.1007, reader_cost: 0.00006, ips: 39.7024 samples/sec | ETA 00:01:22
2022-05-19 10:37:01 [INFO]      [TRAIN] epoch: 3, iter: 190/1000, loss: 0.0005, lr: 0.041408, batch_cost: 0.1005, reader_cost: 0.00006, ips: 39.7975 samples/sec | ETA 00:01:21
2022-05-19 10:37:02 [INFO]      [TRAIN] epoch: 3, iter: 200/1000, loss: 0.0006, lr: 0.040949, batch_cost: 0.1032, reader_cost: 0.00006, ips: 38.7768 samples/sec | ETA 00:01:22
2022-05-19 10:37:02 [INFO]      Start evaluating (total_samples: 237, total_iters: 237)...
237/237 [==============================] - 7s 31ms/step - batch_cost: 0.0309 - reader cost: 1.6418e-04
2022-05-19 10:37:09 [INFO]      [EVAL] #Images: 237 mIoU: 0.5000 Acc: 1.0000 Kappa: nan Dice: 0.5000
2022-05-19 10:37:09 [INFO]      [EVAL] Class IoU: 
[1. 0.]
2022-05-19 10:37:09 [INFO]      [EVAL] Class Precision: 
[1. 0.]
2022-05-19 10:37:09 [INFO]      [EVAL] Class Recall: 
[1. 0.]

**Comment by caomao1111** [comment]
log文件有点奇怪，是一堆16进制数
1a00 0000 0000 0000 0a18 120d 6d65 7461
5f64 6174 615f 7461 6718 82bd f2a8 8d30
5200 1c00 0000 0000 0000 0a1a 080a 120a
5472 6169 6e2f 6c6f 7373 18ab d0f2 a88d
3025 b56a 0b40 1e00 0000 0000 0000 0a1c
080a 120c 5472 6169 6e2f 6c6f 7373 5f30
18ab d0f2 a88d 3025 6ed5 b13e 1e00 0000
0000 0000 0a1c 080a 120c 5472 6169 6e2f
6c6f 7373 5f31 18ab d0f2 a88d 3025 6639
813f 1e00 0000 0000 0000 0a1c 080a 120c
5472 6169 6e2f 6c6f 7373 5f32 18ab d0f2
a88d 3025 a607 9a3e 1e00 0000 0000 0000
0a1c 080a 120c 5472 6169 6e2f 6c6f 7373
5f33 18ab d0f2 a88d 3025 0a9a ad3e 1e00
0000 0000 0000 0a1c 080a 120c 5472 6169
6e2f 6c6f 7373 5f34 18ab d0f2 a88d 3025
edf1 393e 1a00 0000 0000 0000 0a18 080a
1208 5472 6169 6e2f 6c72 18ab d0f2 a88d
3025 2683 223c 2200 0000 0000 0000 0a20
080a 1210 5472 6169 6e2f 6261 7463 685f
636f 7374 18ab d0f2 a88d 3025 8631 7d3e
2300 0000 0000 0000 0a21 080a 1211 5472
6169 6e2f 7265 6164 6572 5f63 6f73 7418
abd0 f2a8 8d30 2540 2dcb 3c1c 0000 0000
0000 000a 1a08 1412 0a54 7261 696e 2f6c
6f73 7318 ced9 f2a8 8d30 25ed 7da7 3e1e
0000 0000 0000 000a 1c08 1412 0c54 7261
696e 2f6c 6f73 735f 3018 ced9 f2a8 8d30
2597 ae9f 3b1e 0000 0000 0000 000a 1c08
1412 0c54 7261 696e 2f6c 6f73 735f 3118
ced9 f2a8 8d30 2595 7851 3e1e 0000 0000
0000 000a 1c08 1412 0c54 7261 696e 2f6c
6f73 735f 3218 ced9 f2a8 8d30 25c2 32a6
3d1e 0000 0000 0000 000a 1c08 1412 0c54
7261 696e 2f6c 6f73 735f 3318 ced9 f2a8
8d30 2568 74e0 3c1e 0000 0000 0000 000a
1c08 1412 0c54 7261 696e 2f6c 6f73 735f
3418 ced9 f2a8 8d30 2532 de15 3c1a 0000
0000 0000 000a 1808 1412 0854 7261 696e
2f6c 7218 ced9 f2a8 8d30 2521 0921 3c22
0000 0000 0000 000a 2008 1412 1054 7261
696e 2f62 6174 6368 5f63 6f73 7418 ced9
f2a8 8d30 25b0 1ef3 3d23 0000 0000 0000
000a 2108 1412 1154 7261 696e 2f72 6561
6465 725f 636f 7374 18ce d9f2 a88d 3025
00b0 cc39 1c00 0000 0000 0000 0a1a 081e
120a 5472 6169 6e2f 6c6f 7373 18f1 e2f2
a88d 3025 65b3 c23d 1e00 0000 0000 0000
0a1c 081e 120c 5472 6169 6e2f 6c6f 7373
5f30 18f1 e2f2 a88d 3025 d5bd 803a 1e00
0000 0000 0000 0a1c 081e 120c 5472 6169
6e2f 6c6f 7373 5f31 18f1 e2f2 a88d 3025
c094 693d 1e00 0000 0000 0000 0a1c 081e
120c 5472 6169 6e2f 6c6f 7373 5f32 18f1
e2f2 a88d 3025 f2f9 ec3c 1e00 0000 0000
0000 0a1c 081e 120c 5472 6169 6e2f 6c6f
7373 5f33 18f1 e2f2 a88d 3025 4bdf c43b
1e00 0000 0000 0000 0a1c 081e 120c 5472
6169 6e2f 6c6f 7373 5f34 18f1 e2f2 a88d
3025 9b33 0b3b 1a00 0000 0000 0000 0a18
081e 1208 5472 6169 6e2f 6c72 18f1 e2f2
a88d 3025 b98e 1f3c 2200 0000 0000 0000
0a20 081e 1210 5472 6169 6e2f 6261 7463
685f 636f 7374 18f1 e2f2 a88d 3025 4aef
f23d 2300 0000 0000 0000 0a21 081e 1211
5472 6169 6e2f 7265 6164 6572 5f63 6f73
7418 f1e2 f2a8 8d30 2533 b3b1 381c 0000
0000 0000 000a 1a08 2812 0a54 7261 696e
2f6c 6f73 7318 a2ec f2a8 8d30 2575 3c3c
3d1e 0000 0000 0000 000a 1c08 2812 0c54
7261 696e 2f6c 6f73 735f 3018 a2ec f2a8
8d30 259b 18f7 391e 0000 0000 0000 000a
1c08 2812 0c54 7261 696e 2f6c 6f73 735f
3118 a2ec f2a8 8d30 2508 90d4 3c1e 0000
0000 0000 000a 1c08 2812 0c54 7261 696e
2f6c 6f73 735f 3218 a2ec f2a8 8d30 253d
c577 3c1e 0000 0000 0000 000a 1c08 2812
0c54 7261 696e 2f6c 6f73 735f 3318 a2ec
f2a8 8d30 2512 1751 3b1e 0000 0000 0000
000a 1c08 2812 0c54 7261 696e 2f6c 6f73
735f 3418 a2ec f2a8 8d30 25ee 6fa0 3a1a
0000 0000 0000 000a 1808 2812 0854 7261
696e 2f6c 7218 a2ec f2a8 8d30 25ed 131e
3c22 0000 0000 0000 000a 2008 2812 1054
7261 696e 2f62 6174 6368 5f63 6f73 7418
a2ec f2a8 8d30 25ad e4f5 3d23 0000 0000
0000 000a 2108 2812 1154 7261 696e 2f72
6561 6465 725f 636f 7374 18a2 ecf2 a88d
3025 6636 cc39 1c00 0000 0000 0000 0a1a
0832 120a 5472 6169 6e2f 6c6f 7373 18a0
f7f2 a88d 3025 d484 0a3d 1e00 0000 0000
0000 0a1c 0832 120c 5472 6169 6e2f 6c6f
7373 5f30 18a0 f7f2 a88d 3025 9034 ea39
1e00 0000 0000 0000 0a1c 0832 120c 5472
6169 6e2f 6c6f 7373 5f31 18a0 f7f2 a88d
3025 c90e 943c 1e00 0000 0000 0000 0a1c
0832 120c 5472 6169 6e2f 6c6f 7373 5f32
18a0 f7f2 a88d 3025 e960 403c 1e00 0000
0000 0000 0a1c 0832 120c 5472 6169 6e2f
6c6f 7373 5f33 18a0 f7f2 a88d 3025 3634
293b 1e00 0000 0000 0000 0a1c 0832 120c
5472 6169 6e2f 6c6f 7373 5f34 18a0 f7f2
a88d 3025 1562 7f3a 1a00 0000 0000 0000
0a18 0832 1208 5472 6169 6e2f 6c72 18a0
f7f2 a88d 3025 bd98 1c3c 2200 0000 0000

**Comment by caomao1111** [comment]
本数据集是二分类的（背景与裂缝两类），进行分类时所有像素都被分为背景类，训练完全没有起作用

**Comment by caomao1111** [comment]
数据集配置
train_dataset:
  type: Dataset
  dataset_root: data/DeepCrack
  train_path: data/DeepCrack/train_list2.txt
  num_classes: 2
  transforms:
    - type: Resize
      target_size: [544,384]
    - type: RandomHorizontalFlip
    - type: Normalize
  mode: train

**Comment by shiyutang** [comment]
It seems the loss does not take effect because it is tiny. And it can be told from the log, the accuracy is close to 1. Therefore, we can imagine the dataset is extremely imbalanced, so you should try other losses like Lovasz loss or dice loss, or add weight to the CELoss. And try other method to augment your data.

**Comment by caomao1111** [comment]
我查了下数据集，一张图片里面正负样本比例约为1:30，但我设置weight和改用Lovasz等损失函数Loss依旧太小，Acc依旧一直为1，这是为何，按照全预测为负样本的话Acc也应该为97%左右，为何会直接为1

**Comment by shiyutang** [comment]
I think you can check your validation set and see if there is a problem with the data.
